app/assets/javascripts/discourse/app/lib/highlight-html.js

```diff
@@ -51,7 +52,7 @@ export default function (node, words, opts = {}) {
     return node;
   }
 
-  const pattern = `(${words.join(" ")})`;
+  const pattern = `(\\W|^)(${words.join(" ")})(\\W|$)`;
```

Comment:
This needed to be added to prevent matching word parts (eg, "bar" in "foobar"), but couldn't used `\b` to match word boundaries, because that has poor Unicode support.